pallets/msa/src/migration/v2.rs

```diff
@@ -0,0 +1,160 @@
+use alloc::vec;
+use common_primitives::{
+	msa::{ProviderId, ProviderRegistryEntry},
+	utils::{get_chain_type_by_genesis_hash, DetectedChainType},
+};
+use frame_support::{pallet_prelude::*, traits::OnRuntimeUpgrade, weights::Weight};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_runtime::Saturating;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+
+const LOG_TARGET: &str = "runtime::provider";
+
+/// Get known provider ids from paseo and mainnet
+pub fn get_known_provider_ids<T: Config>() -> vec::Vec<ProviderId> {
+	let genesis_block: BlockNumberFor<T> = 0u32.into();
+	let genesis = <frame_system::Pallet<T>>::block_hash(genesis_block);
+	let detected_chain = get_chain_type_by_genesis_hash(&genesis.encode()[..]);
+	match detected_chain {
+		DetectedChainType::FrequencyPaseoTestNet => vec![
+			// name: PrivateProvider
+			ProviderId::from(5213u64),
+			// name: CapacityProvider
+			ProviderId::from(77_716u64),
+			// name: Provider1
+			ProviderId::from(77_674u64),
+		],
+		DetectedChainType::FrequencyMainNet => vec![
+			// Soar.ai
+			ProviderId::from(1_610_732u64),
+			// AmplicaTest1
+			ProviderId::from(1u64),
+			// MeWe
+			ProviderId::from(2u64),
+		],
+		_ => vec::Vec::new(),
+	}
+}
+
+/// migrate `ProviderToRegistryEntry` translating to new `ProviderRegistryEntry`
+pub struct MigrateProviderToRegistryEntryV2<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for MigrateProviderToRegistryEntryV2<T> {
+	fn on_runtime_upgrade() -> Weight {
+		let onchain_version = Pallet::<T>::on_chain_storage_version();
+		let current_version = Pallet::<T>::in_code_storage_version();
+		log::info!(target: LOG_TARGET, "onchain_version= {onchain_version:?}, current_version={current_version:?}");
+		if STORAGE_VERSION != current_version {
+			log::error!(target: LOG_TARGET, "Storage version mismatch. Expected: {STORAGE_VERSION:?}, Found: {current_version:?}");
+			return T::DbWeight::get().reads(1)
+		}
+		if onchain_version < current_version {
+			log::info!(target: LOG_TARGET, "Migrating v1 ProviderToRegistryEntry to updated ProviderRegistryEntry...");
+
+			// Count items in OLD storage using storage alias
+			let total_count = v1::ProviderToRegistryEntry::<T>::iter().count();
+			log::info!(target: LOG_TARGET, "Total items in OLD ProviderToRegistryEntry storage: {total_count}");
+			// We expect to have providers in the old storage
+			if total_count == 0 {
+				log::error!(target: LOG_TARGET, "No items found in OLD ProviderToRegistryEntry storage, skipping migration.");
+				return T::DbWeight::get().reads(1)
+			}
+			let mut reads = 1u64;
+			let mut writes = 0u64;
+			let mut bytes = 0u64;
+			let mut migrated_count = 0u64;
+
+			log::info!(target: LOG_TARGET, "Starting migration from v1 to v2...");
+			// Drain from OLD storage and migrate to NEW storage
+			for (provider_id, old_entry) in v1::ProviderToRegistryEntry::<T>::drain() {
```

Comment:
This is not going to work for paseo. We have around 3635 items which will not fit in a block. We need multi-block migration to make it work on paseo

Reply:
Do we have a defined way to run multi-block migration?

Reply:
We might want to discuss this at midweek. My understanding was that Paseo was intended to be reset periodically. If that's an option, it seems a waste of time to do the work to code a multi-block migration only to support the state on testnet. Questions to be answered:
* Is it possible to "reset" Paseo? Level of effort?
    * What services could be affected by a reset? (Freesky, CI pipelines, FA, Provider Dashboard)
    * Would the effort involved in resetting Paseo be worth it vs. coding a more complex migration?
* Is it possible to simply prune the Provider storage on Paseo using a sudo script? Level of effort?
    * What would happen to staked tokens on testnet if an MSA is suddenly "de-authorized" as a Provider?
    * Same as above: worth the effort vs multi-block migration?